app/Main.hs

```diff
@@ -116,9 +116,33 @@ main = do setLocaleEncoding utf8
               preloads = optPreload fullOpts
               postloads = optPostload fullOpts
               load = flip loadFiles
+              setExecutionMode Repl ctx =
+                execStr "Internal" ("(defmodule Introspect (defdynamic execution-mode-dynamic 'Repl))") ctx
```

Comment:
I'm afraid I still don't get why these are defined by evaluating strings here, rather than in a file or as built in commands?

Reply:
Oh, I see what you mean now--sorry, I misunderstood your original comment!

We can define these as commands, I think it's more work to plumb the execution mode into the command code but it's certainly possible. There's no good reason for executing these as strings other than it was simpler than plumbing the Execution mode through to a command--but I think its carried through the context? So that should work.

@hellerve suggested this as well, I'll take some more time on this and impl these as commands.